scripts/integration/README.md

```diff
@@ -0,0 +1,28 @@
+This directory contains a set of integration test frameworks for vector which are executed by the
+`vdev` tool.
+
+Each directory contains two files:
+
+1. A `compose.yaml` file containing the instructions to `docker-compose` or `podman-compose` for how
+   to set up the containers in which to run the integrations, and
+2. A `test.yaml` file that describes how to run the integration tests, including a matrix of
+   software versions or other parameters over which the tests will be run.
+
+You can list these tests with `cargo vdev integration show`[1], which provides a list of all the
+integration test names followed by the extrapolated matrix of environments.
+
+Each test can be run using one of the following:
+
+1. Run a single test environment from the above list with `cargo vdev integration test NAME ENV`
+2. Run all the environments for one test with `cargo vdev integration test NAME`
+3. Run all the steps individually using the `start`, `test`, and then `stop` subcommands with the
+   same parameters as above:
+
+```shell
+vdev integration start NAME ENVIRONMENT
+vdev integration test NAME ENVIRONMENT
+vdev integration stop NAME
```

Comment:
I haven't inspected the code deeply yet, but I pulled the PR and just ran the `show` command.
Maybe I'm being dense but it isn't super clear to me what the "environment" is? Is it just a container version?

Also, I had to read this twice to discern that the `cargo` part of the command strings together the start, test, and stop subcommands of vdev for you. Might be worth calling out?

Reply:
The "environment" is a little… flexible. It is a product of all the fields listed in the `matrix` of the `test.yaml` file. For most of them (all currently?), it is the version or tag of the container, but that is not necessarily its only use.

Reply:
Perhaps there a better term than environment we could think up?

Reply:
I suppose keeping the name "environment" is ok, but maybe we could add a note in the README wrt to the clarification Bruce just provided. Since just looking at the output of the show command, and seeing a jumble of numbers and "latest" , but all currently having a single "environment" could be puzzling. It was for me at least.

```
Integration Name  Environment Name(s)
----------------  -------------------
amqp              3.8
apex              latest
aws               latest
axiom             13-alpine
azure             3.14.0
chronicle         latest
clickhouse        19
datadog-agent     7 (active)
datadog-traces    latest
elasticsearch     7.13.1
eventstoredb      latest
gcp               latest
http-client       latest
humio             1.13.1
influxdb          1.8
kafka             latest
logstash          7.12.1
loki              2.4.1
mongodb           4.2.10
nats              latest
nginx             1.19.4
opentelemetry     0.56.0
postgres          13.1
prometheus        1.8
pulsar            latest
redis             6-alpine
shutdown          latest
splunk            8.2.4
```

Reply:
Seems that Bruce is saying that isn't always necessarily going to be the case though.